awswrangler/s3/_select.py

```diff
@@ -18,26 +18,32 @@
 _RANGE_CHUNK_SIZE: int = int(1024 * 1024)
 
 
+def _gen_scan_range(obj_size: int) -> Iterator[Tuple[int, int]]:
+    for i in range(0, obj_size, _RANGE_CHUNK_SIZE):
+        yield (i, i + min(_RANGE_CHUNK_SIZE, obj_size - i))
+
+
 def _select_object_content(
     args: Dict[str, Any],
     client_s3: boto3.Session,
     scan_range: Optional[Tuple[int, int]] = None,
 ) -> pd.DataFrame:
     if scan_range:
-        args.update({"ScanRange": {"Start": scan_range[0], "End": scan_range[1]}})
-    response = client_s3.select_object_content(**args)
+        response = client_s3.select_object_content(**args, ScanRange={"Start": scan_range[0], "End": scan_range[1]})
+    else:
+        response = client_s3.select_object_content(**args)
 
     dfs: List[pd.DataFrame] = []
     partial_record: str = ""
     for event in response["Payload"]:
         if "Records" in event:
-            records = partial_record + event["Records"]["Payload"].decode(encoding="utf-8", errors="ignore")
-            split_records = records.split("\n")
+            records = event["Records"]["Payload"].decode(encoding="utf-8", errors="ignore").split("\n")
+            records[0] = partial_record + records[0]
```

Comment:
Perhaps joining the records here in sudden case there will be more than one in the future?